### ISSUE TITLE
[Bug]: Netlify doesn't load middleware when project is located in /src

### ISSUE THREAD

**Issue by jimmi-rbn** [netlify/next-runtime#1746]
### Summary

I expect Netlify to load middleware when deployed.

It only works locally with `netlify dev`

### Steps to reproduce

Create next project `npx create-next-app@latest`
Create simple middleware.js
Test locally with `netlify dev`

Deploy to netlify

### A link to a reproduction repository

https://github.com/jimmi-rbn/nextjsMiddleware

### Next Runtime version

12.3.0

### More information about your build

- [ ] I am building using the CLI
- [ ] I am building using file-based configuration (`netlify.toml`)

### What OS are you using?

Mac OS

### Your netlify.toml file

<details>
  <summary>`netlify.toml`</summary>

```toml
# Paste content of your `netlify.toml` file here
```

</details>

### Your public/_redirects file

<details>
  <summary>`_redirects`</summary>

```toml
# Paste content of your `_redirects` file here
```

</details>

### Your `next.config.js` file

<details>
  <summary>`next.config.js`</summary>

```toml
# Paste content of your `next.config.js` file here. Check there is no private info in there.
```

</details>

### Builds logs (or link to your logs)

<details>
  <summary>Build logs</summary>

```
# Paste logs here
```

</details>

### Function logs

<details>
  <summary>Function logs</summary>

```
# Paste logs here
```

</details>

### .next JSON files

<details>
  <summary>generated .next JSON files</summary>

```
# Paste file contents here. Please check there isn't any private info in them
# You can either build locally, or download the deploy from Netlify by clicking the arrow next to the deploy time.
```

</details>

**Comment by nickytonline** [comment]
Hi @jimmi-rbn, thanks for opening the issue. Like you mentioned, it should work if the middleware is in the root of the project or the root of the `src` folder. It looks like at the moment, we only support it in the root of the project. I'm going to add this to our backlog, but in the meantime, if you put it in the root of your Next project, you should be all good for now.

**Comment by orinokai** [comment]
Looks like this is a recent Next.js change https://github.com/vercel/next.js/pull/40340/files that needs implementing

**Comment by ascorbic** [comment]
Hi. I think this is an issue with how you've set up your project. You've moved the pages into `src`, but not the middleware. I checked-out your repo and ran a local build without the Netlify CLI. I then looked at `.next/server/middleware-manifest.json`, and there was no middleware shown. If I moved the midldeware into `src` then it started working. I think the bug is actually with `netlify dev`. It _shouldn't_ be working with middleware in the root and pages in `src`. There is a separate issue where it doesn't work with `netlify dev` middleware in `src`, but that should be fixed soon.
